tsdb/index/tsi1/index.go

```diff
@@ -722,7 +725,10 @@ func (i *Index) CreateSeriesListIfNotExists(keys [][]byte, names [][]byte, tagsS
 		}
 	}
 
-	// Update sketches.
+	// Update sketches under lock.
+	i.mu.Lock()
```

Comment:
I'm a little concerned about locking here. Is this the only way we can safely update the sketches?